Frends.AmazonS3.DownloadObject/Frends.AmazonS3.DownloadObject.Tests/UnitTest1.cs

```diff
@@ -0,0 +1,554 @@
+        public void PreSignedMissingDestinationPathTest()
+        {
+            var setS3Key = $"2022/Testfile.txt"; //S3 key.
+
+            _connection = new Connection()
+            {
+                AuthenticationMethod = AuthenticationMethod.PreSignedURL,
+                PreSignedURL = CreatePreSignedURL(setS3Key)
+            };
+
+            _input = new Input()
+            {
+                DestinationPath = "",
+                FileName = "Testfile.txt",
+            };
+
+            _options = new Options() { };
+
+            var ex = Assert.ThrowsExceptionAsync<Exception>(async () => await AmazonS3.DownloadObject(_connection, _input, _options, default)).Result;
+            Assert.IsTrue(ex.Message.Contains("Destination required."));
+        }
+
+        /// <summary>
+        /// File already exists in destination. Overwrite and ContinueIfExists = false.
+        /// </summary>
+        [TestMethod]
+        public void PreSignedNoOverwriteContinueIfExistsTest()
+        {
+            var setS3Key = $"2022/Overwrite.txt"; //S3 key.
+
+            _connection = new Connection()
+            {
+                AuthenticationMethod = AuthenticationMethod.PreSignedURL,
+                PreSignedURL = CreatePreSignedURL(setS3Key)
+            };
+
+            _input = new Input()
+            {
+                DestinationPath = @$"{_dir}\Download",
+                FileName = "Overwrite.txt",
+            };
+
+            _options = new Options()
+            {
+                Overwrite = false,
+                ContinueIfExists = false,
+            };
+
+            var ex = Assert.ThrowsExceptionAsync<Exception>(async () => await AmazonS3.DownloadObject(_connection, _input, _options, default)).Result;
+            Assert.IsTrue(ex.Message.Contains("File already exists at"));
+        }
+
+        /// <summary>
+        /// Result is not null, file exists and was overwriten (compare lines).
+        /// </summary>
+        [TestMethod]
+        public void PreSignedURLOverwriteTest()
+        {
+            var setS3Key = $"2022/Overwrite.txt"; //S3 key.
+
+            _connection = new Connection()
+            {
+                AuthenticationMethod = AuthenticationMethod.PreSignedURL,
+                PreSignedURL = CreatePreSignedURL(setS3Key)
+            };
+
+            _input = new Input()
+            {
+                DestinationPath = @$"{_dir}\Download",
+                FileName = "Overwrite.txt",
+            };
+
+            _options = new Options()
+            {
+                Overwrite = true,
+                ContinueIfExists = false,
+            };
+
+            var result = AmazonS3.DownloadObject(_connection, _input, _options, default);
+            Assert.IsNotNull(result.Result.Results);
+            Assert.IsTrue(File.Exists(@$"{_dir}\Download\Overwrite.txt"));
+            Assert.IsTrue(CompareFiles());
+        }
+
+        #endregion pre-signed URL
+
+        #region aws creds
```

Comment:
No need for region here.

Reply:
Tests are in 2 parts, pre-signed and AWS creds, so it's easier to follow the code when you can collapse pre-signed part out of sight. This doesn't affect the code.